src/Hosting/TestHost/test/WebSocketClientTests.cs

```diff
@@ -54,5 +55,48 @@ public async Task ConnectAsync_ShouldSetRequestProperties(string requestUri, str
             Assert.Equal(expectedHost, capturedHost);
             Assert.Equal("/connect", capturedPath);
         }
+
+        [Fact]
+        public async Task CanAcceptWebSocket()
```

Comment:
You want a different test for the normal TestServer scenario that shows the upgrade and WebSocket features are present, but an upgrade isn't possible.